### PR TITLE
Prepared branch for Jersey 3.0.0-M3 release

### PR DESCRIPTION
**Pull request by senivam**
Branch aligned with master (1 commit before 2.31 release), contains  jakartified dependencies for :

jsp, jstl, jaxb-api/ri (bumped versions for latest Mx versions), json-b/p, moxy, persistence, istack.commons, EJB, Interceptor, Yasson

this PR is created as reduced (number of commits) version of #4468. Which is being closed in favor of this PR.

### CODE REVIEW

File: ext/mvc-jsp/pom.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
                 <configuration>
                     <instructions>
                         <Import-Package>
-                            javax.servlet.jsp.*;version="[2.0,3.0)",
+                            jakarta.servlet.jsp.*;version="[2.0,3.0)",
```
Comment (jansupol) [inline ext/mvc-jsp/pom.xml:56]: This is a nonsense

File: media/json-processing/pom.xml [hunk 2]
```diff
@@ -57,14 +57,25 @@
                         <!-- TODO: change to ${range;[==,+);${jsonp.api.version}} once the version is final-->
                         <Import-Package>
                             ${jakarta.annotation.osgi.version},
-                            javax.json.*;version="[0,2)",
-                            org.glassfish.json.*;version="[0,2)",
+                            jakarta.json.*;version="[0,2)",
```
Comment (jansupol) [inline media/json-processing/pom.xml:60]: this will not work in OSGi

File: ext/mvc-jsp/pom.xml (hunk 1 above)
Comment (senivam) [reply ext/mvc-jsp/pom.xml]: upper version fixed

File: media/json-processing/pom.xml (hunk 2 above)
Comment (senivam) [reply media/json-processing/pom.xml]: upper version fixed

**Comment by arjantijms** [comment]
@jansupol @senivam I haven't looked at the recent state of the code in detail, but maybe reverting this commit fixes all problems?

https://github.com/eclipse-ee4j/jersey/pull/4468/commits/907e5322019076b6b0032faddcabace8016b19be

I think we need to Jakartify, not de-Jakartify, as that is what we already have now. In the GlassFish project we're waiting for a Jakartified M3. Sorry if I have misunderstood the commit, but at glance it seems something is confused there.

**Comment by senivam** [comment]
@arjantijms no, the commit is inteneded to be retracted :) When grizzly and servlet api will be ready (available in staging with correct functionallitiy) it will be retracted. I've intentionnaly placed all de-jakartification in one commit to easily revert it when time comes.

**Comment by arjantijms** [comment]
@senivam I'm very close to staging Grizzly. The CI finally builds and hopefully we get the permission soon from Sonatype to actually stage.

**Comment by senivam** [comment]
@arjantijms, @jansupol   Jersey 3.0.0-M3 is staged. Servlets 5.0.0-M1 and Grizzly 3.0.0-M1 presumably will be included into Jersey 3.0.0-M4

**Comment by arjantijms** [comment]
@senivam Thanks. We will not be using 3.0.0-M3, since without Servlets 5 and Grizzly 3 it would be useless to us (it bails out immediately with OSGi failures). But we will wait for M4 then.

**Comment by arjantijms** [comment]
p.s. alternatively I can also stage a 3.0.0-M4 myself if you don't mind. Stagings are cheap anyway and can be removed at will if needed.

**Comment by senivam** [comment]
@arjantijms Hi Arjan, I understand you are in hurry, and I suppose, the M4th version will be available ASAP. I'm working on the M4th version already, it will contain (besides servlets and Grizzly) several merges and will be aligned with actual master. Please let me know when Grizzly is staged.

**Pull request merged**